UI: fix serviceName ifs for YouTube services

mv2731 opened the pull request:
Update the if statements in OBSBasicSettings method
to include the updated name of YouTube / YouTube Gaming
and to check for Youtube-HLS

mv2731 committed:
UI: Update Youtube get stream key URL

The old URL is going to be deprecated soon, and this is why we replaced
with a new one.

mv2731 committed:
UI: Added www to Youtube stream key URL

Added www to be beginning of the stream key URL we are using to replace
the deprecated live_dashboard Youtube stream key link.

mv2731 committed:
UI: Fix serviceName ifs for YouTube services

Update the if statements in OBSBasicSettings method
to include the updated name of YouTube / YouTube Gaming
and to check for Youtube-HLS

mv2731 committed:
UI: Fix missing parenthesis in serviceName ifs

ushadow commented on UI/window-basic-auto-config.cpp:
```diff
@@ -566,7 +566,8 @@ void AutoConfigStreamPage::UpdateKeyLink()
 		streamKeyLink =
 			"https://www.twitch.tv/broadcast/dashboard/streamkey";
-	} else if (serviceName == "YouTube / YouTube Gaming") {
+	} else if ((serviceName == "YouTube-RTMP") or
```
This change needs to be pushed to add-youtube-hls-streaming branch, and keep the link to the old url.

ushadow replied:
Please add spaces before and after "-", i.e. YouTube - RTMP, to be consistent with the names for other services. Same for HLS, and also need to change in services.json.

mv2731 committed:
UI: Add spaces to the serviceNames

Change YouTube service names "YouTube-HLS" to "Youtube - HLS"
for better consistency with other service names in the product
and revert our changes replacing the deprecated live_dashboard
link.

ushadow commented on UI/window-basic-auto-config.cpp:
```diff
@@ -566,7 +566,8 @@ void AutoConfigStreamPage::UpdateKeyLink()
 		streamKeyLink =
 			"https://www.twitch.tv/broadcast/dashboard/streamkey";
-	} else if (serviceName == "YouTube / YouTube Gaming") {
+	} else if ((serviceName == "YouTube - RTMP") or
```
Checking the codebase, I think the convention is using || instead of the keyword or. And same below.

ushadow commented on UI/window-basic-auto-config.cpp:
```diff
@@ -566,7 +566,8 @@ void AutoConfigStreamPage::UpdateKeyLink()
 		streamKeyLink =
 			"https://www.twitch.tv/broadcast/dashboard/streamkey";
-	} else if (serviceName == "YouTube / YouTube Gaming") {
+	} else if ((serviceName == "YouTube - RTMP") or
```
Could you change or to || for consistency with the rest of the code base? And same below.

ushadow replied:
Are the service names in services.json file also updated with spaces around -? Are the changes not in this commit? It's fine if they will be in a different commit. Just checking.

ushadow reviewed (changes requested):
See review comments.

mv2731 committed:
UI: Change or into ||

Change or keyword to || to be more consistent
with the rest of the codebase. Also change Youtube service
names in services.json to have consistent spacing.

mv2731 committed:
Merge branch 'add-youtube-hls-streaming' into update-youtube-url-wip

ushadow commented on plugins/rtmp-services/data/services.json:
```diff
@@ -198,7 +198,8 @@
         },
         {
-            "name": "YouTube-HLS",
+
```
Extra empty line.

mv2731 committed:
plugins: Remove blank line
